packages/wrangler/src/dev.tsx

```diff
@@ -932,18 +976,33 @@ export async function startDev(args: StartDevOptions) {
 		}
 
 		const devReactElement = render(await getDevReactElement(config));
-
 		rerender = devReactElement.rerender;
+
+		if (experimentalAssets && !args.experimentalDevEnv) {
+			assetsWatcher = watch(experimentalAssets.directory, {
+				persistent: true,
+				ignoreInitial: true,
+			}).on("all", async (eventName, filePath) => {
+				const message = getAssetChangeMessage(eventName, filePath);
+
+				logger.log(`🌀 ${message}...`);
+				rerender(await getDevReactElement(config));
+			});
+		}
+
 		return {
 			devReactElement,
-			watcher,
+			configFileWatcher,
+			assetsWatcher,
 			stop: async () => {
 				devReactElement.unmount();
-				await watcher?.close();
+				await configFileWatcher?.close();
+				await assetsWatcher?.close();
 			},
 		};
 	} catch (e) {
-		await watcher?.close();
+		await configFileWatcher?.close();
+		await assetsWatcher?.close();
```

Comment:
If the first call rejects, the 2nd won't be called.

We should wrap these in Promise.allSettled

Reply:
✅